File: Leetcode Daily Challenge/032-February-2026/005-transformed-array.py
# https://leetcode.com/problems/transformed-array/




from typing import List

class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n
        for i in range(n):
            val = nums[i]
            if nums[i] > 0:
                val = nums[(i + nums[i]) % n] 
            elif nums[i] < 0:
                val = nums[(i - (abs(nums[i]) % n) + n) % n]
            result[i] = val

        return result
            
        

File: Leetcode Daily Challenge/032-February-2026/005-transformed-array.java
// https://leetcode.com/problems/transformed-array/




class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        for (int i = 0; i < n; i ++) {
            int val = nums[i];
            if (nums[i] > 0) val = nums[(i + nums[i]) % n];
            else if (nums[i] < 0) val = nums[(i - Math.abs(nums[i]) % n + n) % n];
            result[i] = val;
        }
        return result;
    }
}